napari/_qt/widgets/qt_keyboard_settings.py

```diff
@@ -581,7 +581,22 @@ def event(self, event):
         if event.type() == QEvent.Type.ShortcutOverride:
             self.keyPressEvent(event)
             return True
-        if event.type() in [QEvent.Type.KeyPress, QEvent.Type.Shortcut]:
+        if event.type() in [
+            QEvent.Type.KeyPress,
+            QEvent.Type.Shortcut,
+        ] and event.key() not in (
+            Qt.Key.Key_Control,
+            Qt.Key.Key_Shift,
+            Qt.Key.Key_Alt,
+            Qt.Key.Key_Meta,
+        ):
+            # Need to let pass events for modifier keys in order to be able to
+            # set them as shortcuts on some OS/binding combinations
+            # (for example macOS + PyQt6).
+            # Only single modifiers are valid as shortcuts, for more context
+            # see `ShortcutEditor._set_keybinding`.
+            # Other keys/shortcut events apparently don't need this handling
+            # and can be marked as recognized and processed (returning True)
```

Comment:
I do not understand this comment. Maybe you should above if condition, as the comment suggests that there is a need to pass `event` to `keyPressEvent` function.